core/src/main/java/nu/yona/server/device/entities/DeviceAnonymized.java

```diff
@@ -148,4 +153,14 @@ public void addVpnStatusChangeEvent(VpnStatusChangeEvent event)
 	{
 		return Optional.ofNullable(lastVpnStatusChangeEvent);
 	}
+
+	public Optional<String> getFirebaseInstanceId()
+	{
+		return Optional.ofNullable(firebaseInstanceId);
+	}
+
+	public void setFirebaseInstanceId(Optional<String> firebaseInstanceId)
```

Comment:
Should we allow clearing it? I don't think so.

Reply:
I think it is possible. See [onTokenRefresh documentation](https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceIdService.html#onTokenRefresh()): some of those could apply, isn't it?

Reply:
Correction: [onNewToken](https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService#onNewToken(java.lang.String)) since June 28. So maybe Google now thinks it is never cleared. I'm not entirely sure if there could be a situation in which the app would like to revoke the token.

Reply:
I favor going for simple at first. If we run into NPEs, we can enhance it easily.